### PR TITLE
Extracting documents related stuff from the reducer

### PR DESCRIPTION
**Pull request by tejans24**
connects #2829 

Note: This got slightly larger than 200 line limit, it has to do with code being moved around.

1). This is the 1st step of splitting documents away from the reducer
2). The plan is to have a DocumentsList reducer that is used for DocumentsList/Table components. These changes will be moved to that reducer in the future.

## AC
* Open DocumentsList page in Reader
* Open a few documents, make sure the documents load and actions can be performed on the documents; such as rotating, setting categories, etc
* Things should work as expected

### CODE REVIEW

File: client/app/reader/DocumentList/DocumentsReducer.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash';
+
+import * as Constants from '../constants';
+import { update } from '../../util/ReducerUtil';
+
+const documentsReducer = (state = {}, action = {}) => {
+  switch (action.type) {
+  case Constants.RECEIVE_DOCUMENTS:
```
Comment (mdbenjam) [inline client/app/reader/DocumentList/DocumentsReducer.js]: This looks to be in two places.

File: client/app/reader/reducer.js [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 import * as Constants from './constants';
+
 import _ from 'lodash';
 import { update } from '../util/ReducerUtil';
 import { categoryFieldNameOfCategoryName, moveModel } from './utils';
 import { searchString, commentContainsWords, categoryContainsWords } from './search';
 import { timeFunction } from '../util/PerfDebug';
+import documentsReducer from './DocumentList/DocumentsReducer';
+
+const updateDocuments = (state, action) => update(state, {
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Why aren't we just combining reducers?

Comment (tejans24) [reply client/app/reader/reducer.js]: That was going to be a part of the next step, after the documents have been completely separated logically.

File: client/app/reader/DocumentList/DocumentsReducer.js (hunk 1 above)
Comment (tejans24) [reply client/app/reader/DocumentList/DocumentsReducer.js]: `RECEIVE_DOCUMENTS` is being used in both reducer and DocumentsReducer. The one in reducer is calling the documentsReducer function
 
```
documents: {
   $set: documentsReducer(state.documents, action)
}
```

File: client/app/reader/commonActions.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+export const hideErrorMessage = (messageType) => ({
+  type: HIDE_ERROR_MESSAGE,
+  payload: {
+    messageType
+  }
+});
+
+export const showErrorMessage = (messageType) => ({
+  type: SHOW_ERROR_MESSAGE,
```
Comment (mdbenjam) [inline client/app/reader/commonActions.js]: I don't see the corresponding reducer.

Comment (tejans24) [reply client/app/reader/commonActions.js]: Good catch, I added that them to the reducer.

**Pull request merged**